packages/core/src/browser/preferences/preference-contribution.ts

```diff
@@ -59,36 +59,45 @@ export function bindPreferenceSchemaProvider(bind: interfaces.Bind): void {
     bindContributionProvider(bind, PreferenceContribution);
 }
 
+export function bindDefaultPreferenceProvider(bind: interfaces.Bind): void {
+    bind(PreferenceProvider).to(PreferenceSchemaProvider).inSingletonScope().whenTargetNamed(PreferenceScope.Default);
+}
+
 @injectable()
 export class PreferenceSchemaProvider extends PreferenceProvider {
 
-    protected readonly combinedSchema: PreferenceSchema = { properties: {} };
     protected readonly preferences: { [name: string]: any } = {};
+    protected readonly combinedSchema: PreferenceSchema = { properties: {} };
     protected validateFunction: Ajv.ValidateFunction;
 
-    constructor(
-        @inject(ContributionProvider) @named(PreferenceContribution)
-        protected readonly preferenceContributions: ContributionProvider<PreferenceContribution>
-    ) {
-        super();
+    @inject(ILogger) protected readonly logger: ILogger;
+
+    @inject(ContributionProvider) @named(PreferenceContribution)
+    protected readonly preferenceContributions: ContributionProvider<PreferenceContribution>;
+
+    @postConstruct()
+    protected init(): void {
         this.preferenceContributions.getContributions().forEach(contrib => {
             this.doSetSchema(contrib.schema);
         });
+        this.combinedSchema.additionalProperties = false;
         this.updateValidate();
         this._ready.resolve();
     }
 
     protected doSetSchema(schema: PreferenceSchema): void {
         const props: string[] = [];
-        for (const property in schema.properties) {
+        for (const property of Object.keys(schema.properties)) {
+            const schemaProps = schema.properties[property];
             if (this.combinedSchema.properties[property]) {
-                console.error('Preference name collision detected in the schema for property: ' + property);
```

Comment:
`console.error` always calls `Logger` under the hood, there is no need to use it explicitly